app/views/training/index.html.haml

```diff
@@ -3,20 +3,27 @@
   %h1 Training Libraries
   %ul.training-libraries.no-bullets.no-margin
     - defocus_class = @focused_library_slug ? 'training-library-defocus' : ''
-    - @libraries.each do |library|
-      - next if library.exclude_from_index?
-      - focus_class = @focused_library_slug == library.slug ? 'training-library-focus' : defocus_class
-      %li{class: "training-libraries__individual-library no-left-margin #{focus_class}"}
-        %a.action-card.action-card-index{href: "/training/#{library.slug}"}
-          %header.action-card-header
-            %h3.action-card-title
-              = library.translated_name
-            %span.icon-container
-              %i.action-card-icon.icon.icon-rt_arrow
-        .action-card-text
-          %h3 Included Modules:
-          %ul
-            - library.categories.pluck('modules').flatten.each do |training_module|
-              %li
-                %a{href: "/training/#{library.slug}/#{training_module['slug']}", target: "_blank"}
-                  = training_module['name']
+    - unless @libraries.empty?
```

Comment:
I think this is best done in a way that doesn't introduce such a large `if/else` branch in the template, because it is not trivial to skim this file and immediately see what the branch is for.

Instead, I suggest handling this within the controller, using `render(plain: ...) if ...`. That would contain the logic for this change to a single line of code.

Reply:
Ok. As I needed to display a link in one case, I intended to make a method in the controller to be called like
render :html no_training
Problem is message contains a link, so a need to render html and not plain/text. Message display if use html_safe, but htat triggers a warning. I could have use sanitize, but I would have been forced to include ViewHelper ... which is not at its place in a controller.
So I made a new view instead, so that there is only one more line at the end of the index method.

Reply:
These seems like a good solution to me.